### PR TITLE
Cleanup pkgconfig output

### PR DESCRIPTION
**Pull request by folkertdev**

### CODE REVIEW

File: src/pkg_config_gen.rs [hunk 1]
```diff
@@ -307,6 +282,102 @@ mod test {
...
+            let result = canonicalize(path);
+            assert_eq!(result, "/home/user/docs");
+        }
+
+        #[test]
+        fn test_parent_directory() {
+            let path = "/home/user/../docs";
+            let result = canonicalize(path);
+            assert_eq!(result, "/home/user/../docs");
+        }
```
Comment (folkertdev) [inline src/pkg_config_gen.rs:332]: is this expected? should that not output `/home/docs`?

File: src/pkg_config_gen.rs [hunk 2]
```diff
@@ -5,27 +5,31 @@ use crate::install::InstallPaths;
...
+        match component {
             Component::RootDir => {
                 separator = "/";
-                String::new()
             }
-            Component::Prefix(_) => p.as_os_str().to_string_lossy().to_string(),
-            _ => {
-                let c = format!("{}{}", separator, p.as_os_str().to_string_lossy());
+            Component::Prefix(_) => {
+                write!(out, "{}", component_str).unwrap();
```
Comment (folkertdev) [inline src/pkg_config_gen.rs:21]: a write to a `String` will only fail when the system runs out of memory. The rust standard library generally disregards OOM, and it seems reasonable to do here too.

Comment (lu-zero) [reply src/pkg_config_gen.rs:21]: It shouldn't be a problem indeed.

File: src/pkg_config_gen.rs (hunk 1 above)
Comment (lu-zero) [reply src/pkg_config_gen.rs:332]: It should, if you want to add more logic is great :)

**Comment by lu-zero** [comment]
Looks like clippy is now unhappy, willing to add a patch before your set?

File: src/pkg_config_gen.rs (hunk 1 above)
Comment (folkertdev) [reply src/pkg_config_gen.rs:332]: I'll play around with it a bit

Comment (folkertdev) [reply src/pkg_config_gen.rs]: got it to work. less pretty now but more functionality. I think this still gives weird results for e.g. `../` or any case where there are more parent directory components than there are normal components. should that panic maybe? or is that unlikely to come up?

Comment (lu-zero) [reply src/pkg_config_gen.rs]: Its origin is to canonicalize \-paths to /-paths so do not dwell on it your original set + the clippy fix would be already great.

File: src/pkg_config_gen.rs [hunk 3]
```diff
@@ -307,6 +306,102 @@ mod test {
...
+            "Cflags: -I${includedir} -DFOO\n",
+            "Requires: somelib, someotherlib\n",
+            "Requires.private: someprivatelib >= 1.0\n",
+        );
+
+        assert_eq!(expected, pkg.render());
+    }
+
+    mod test_canonicalize {
+        use super::canonicalize;
```
Comment (lu-zero) [inline src/pkg_config_gen.rs:328]: It needs tests for "\a\b\c" and such paths.

Comment (lu-zero) [reply src/pkg_config_gen.rs:328]: But only on windows...

**Review by lu-zero — APPROVED** [review]
It is good enough as it is.

File: src/pkg_config_gen.rs (hunk 3 above)
Comment (folkertdev) [reply src/pkg_config_gen.rs:328]: on windows, `a\b\c` gets parsed as a single normal component. Also pkgconfig is just for unix systems right, so windows paths don' t make a whole lot of sense even when cross-compiling. Maybe WSL causes complications?

Comment (lu-zero) [reply src/pkg_config_gen.rs:328]: pkgconfig exists and is used also on windows :) See https://github.com/lu-zero/cargo-c/issues/210 on why

**Pull request merged**